Failed to build sdl2:x64-linux on fedora

**Describe the bug**
sdl2:x64-linux fails to build, this is using the latest vcpkg

**Environment**
- OS: [Linux]
- Compiler: revision

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install sdl2
2. See error
Repro code when

**Expected behavior**
A clear and concise description of what you expected to happen.

**Failure logs** 
-(please attached failure logs)
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/7464759/config-x64-linux-dbg-err.log)
[config-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/7464760/config-x64-linux-dbg-out.log)

**Additional context**
Add any other context about the problem here.
full command + log here
```
[total@fedora vcpkg]$ ./vcpkg install sdl2
Computing installation plan...
The following packages will be built and installed:
    sdl2[core]:x64-linux -> 2.0.16#1
Detecting compiler hash for triplet x64-linux...
Restored 0 packages from /home/total/.cache/vcpkg/archives in 8.102 us. Use --debug to see more details.
Starting package 1/1: sdl2:x64-linux
Building package sdl2[core]:x64-linux...
-- Using cached libsdl-org-SDL-release-2.0.16.tar.gz.
-- Cleaning sources at /home/total/Documents/vcpkg/buildtrees/sdl2/src/ase-2.0.16-139ee71275.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/total/Documents/vcpkg/downloads/libsdl-org-SDL-release-2.0.16.tar.gz
-- Applying patch 0001-sdl2-Enable-creation-of-pkg-cfg-file-on-windows.patch
-- Applying patch 0002-sdl2-skip-ibus-on-linux.patch
-- Applying patch 0003-sdl2-disable-sdlmain-target-search-on-uwp.patch
-- Applying patch 0004-sdl2-alias-on-static-build.patch
-- Using source at /home/total/Documents/vcpkg/buildtrees/sdl2/src/ase-2.0.16-139ee71275.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /usr/bin/cmake /home/total/Documents/vcpkg/buildtrees/sdl2/src/ase-2.0.16-139ee71275.clean -DVIDEO_VULKAN=OFF -DSDL_STATIC=1 -DSDL_SHARED=0 -DFORCE_STATIC_VCRT=0 -DLIBC=ON -DCMAKE_MAKE_PROGRAM=/home/total/Documents/vcpkg/downloads/tools/ninja-1.10.2-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/total/Documents/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/total/Documents/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/total/Documents/vcpkg -D_VCPKG_INSTALLED_DIR=/home/total/Documents/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/total/Documents/vcpkg/packages/sdl2_x64-linux/debug
    Working Directory: /home/total/Documents/vcpkg/buildtrees/sdl2/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/total/Documents/vcpkg/buildtrees/sdl2/config-x64-linux-dbg-out.log
      /home/total/Documents/vcpkg/buildtrees/sdl2/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:381 (vcpkg_execute_required_process)
  ports/sdl2/portfile.cmake:24 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package sdl2:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: sdl2[core]:x64-linux -> 2.0.16#1
  vcpkg version: 2021-10-25-b4bff9afacafb0a234a8f20fd6451b2362c0f9a0
  vcpkg-tool version: 85b31152d 2021-11-02 (6 hours ago)

Additionally, attach any relevant sections from the log files above.
```

Hi @ITotalJustice Thanks for posting this issue.

```
CMake Error at cmake/macros.cmake:28 (message):
  *** ERROR: Missing Xext.h, maybe you need to install the libxext-dev
  package?
Call Stack (most recent call first):
  cmake/sdlchecks.cmake:444 (message_error)
  CMakeLists.txt:1214 (CheckX11)
```
Seems you installed `x11 `on your machine. Could you please try to install `libxext-dev` to rebuild this?

Of course, we may consider adding `x11 `as a feature for `sdl2`.

> Hi @ITotalJustice Thanks for posting this issue.
> 
> ```
> CMake Error at cmake/macros.cmake:28 (message):
>   *** ERROR: Missing Xext.h, maybe you need to install the libxext-dev
>   package?
> Call Stack (most recent call first):
>   cmake/sdlchecks.cmake:444 (message_error)
>   CMakeLists.txt:1214 (CheckX11)
> ```
> 
> Seems you installed `x11 `on your machine. Could you please try to install `libxext-dev` to rebuild this?
> 
> Of course, we may consider adding `x11 `as a feature for `sdl2`.

Thank you, installed libxext fixed my issue!

For anyone else that comes across this that has this problem on fedora, the command is `dnf install libXext-devel`

--- a/ports/ffmpeg/vcpkg.json
+++ b/ports/ffmpeg/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ffmpeg",
   "version": "4.4",
-  "port-version": 17,
+  "port-version": 18,
   "description": [
     "a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.",
     "FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created. It supports the most obscure ancient formats up to the cutting edge. No matter if they were designed by some standards committee, the community or a corporation. It is also highly portable: FFmpeg compiles, runs, and passes our testing infrastructure FATE across Linux, Mac OS X, Microsoft Windows, the BSDs, Solaris, etc. under a wide variety of build environments, machine architectures, and configurations."
@@ -493,7 +493,18 @@
     "sdl2": {
       "description": "Sdl2 support",
       "dependencies": [
-        "sdl2"
+        {
+          "name": "sdl2",
+          "default-features": false,
+          "features": [
+            "x11"
+          ],
+          "platform": "linux"
+        },
+        {
+          "name": "sdl2",
+          "platform": "!linux"
+        }
       ]
     },
     "snappy": {

--- a/ports/gamedev-framework/vcpkg.json
+++ b/ports/gamedev-framework/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "gamedev-framework",
   "version-semver": "0.20.0",
+  "port-version": 1,
   "maintainers": [
     "Julien Bernard <julien.bernard@univ-fcomte.fr>",
     "Arthur Hugeat <hugeat.arthur@gmail.com>"
@@ -16,7 +17,18 @@
     "boost-heap",
     "freetype",
     "pugixml",
-    "sdl2",
+    {
+      "name": "sdl2",
+      "default-features": false,
+      "features": [
+        "x11"
+      ],
+      "platform": "linux"
+    },
+    {
+      "name": "sdl2",
+      "platform": "!linux"
+    },
     "stb",
     "zlib"
   ]

--- a/ports/pixel/vcpkg.json
+++ b/ports/pixel/vcpkg.json
@@ -1,13 +1,24 @@
 {
   "name": "pixel",
   "version-string": "0.3",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Simple 2D Graphics based on standard and portable OpenGL.",
   "homepage": "https://github.com/dascandy/pixel",
   "supports": "!windows",
   "dependencies": [
     "glew",
     "opengl",
-    "sdl2"
+    {
+      "name": "sdl2",
+      "default-features": false,
+      "features": [
+        "x11"
+      ],
+      "platform": "linux"
+    },
+    {
+      "name": "sdl2",
+      "platform": "!linux"
+    }
   ]
 }

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -26,7 +26,7 @@ if ("x11" IN_LIST FEATURES)
     if (VCPKG_TARGET_IS_WINDOWS)
         message(FATAL_ERROR "Feature x11 only support UNIX.")
     endif()
-    message(WARNING "You will need to install Xorg dependencies to use feature x11:\napt install libx11-dev libxft-dev libxext-dev\n")
+    message(WARNING "You will need to install Xorg dependencies to use feature x11:\nsudo apt install libx11-dev libxft-dev libxext-dev\n")
 endif()
 
 vcpkg_cmake_configure(
@@ -65,7 +65,7 @@ if(NOT BINS)
     )
 endif()
 
-if(VCPKG_TARGET_IS_WINDOWS)
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_UWP)
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
         file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/lib/manual-link")
         file(RENAME "${CURRENT_PACKAGES_DIR}/lib/SDL2main.lib" "${CURRENT_PACKAGES_DIR}/lib/manual-link/SDL2main.lib")

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -18,12 +18,19 @@ string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" FORCE_STATIC_VCRT)
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
-    vulkan  VIDEO_VULKAN
+        vulkan  VIDEO_VULKAN
+        x11     X11_SHARED
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+if ("x11" IN_LIST FEATURES)
+    if (VCPKG_TARGET_IS_WINDOWS)
+        message(FATAL_ERROR "Feature x11 only support UNIX.")
+    endif()
+    message(WARNING "You will need to install Xorg dependencies to use feature x11:\napt install libx11-dev libxft-dev libxext-dev\n")
+endif()
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS ${FEATURE_OPTIONS}
         -DSDL_STATIC=${SDL_STATIC}
         -DSDL_SHARED=${SDL_SHARED}
@@ -31,50 +38,50 @@ vcpkg_configure_cmake(
         -DLIBC=ON
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
 if(EXISTS "${CURRENT_PACKAGES_DIR}/cmake")
-    vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
+    vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
 elseif(EXISTS "${CURRENT_PACKAGES_DIR}/lib/cmake/SDL2")
-    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SDL2)
+    vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/SDL2)
 elseif(EXISTS "${CURRENT_PACKAGES_DIR}/SDL2.framework/Resources")
-    vcpkg_fixup_cmake_targets(CONFIG_PATH SDL2.framework/Resources)
+    vcpkg_cmake_config_fixup(CONFIG_PATH SDL2.framework/Resources)
 endif()
 
 file(REMOVE_RECURSE
-    ${CURRENT_PACKAGES_DIR}/debug/include
-    ${CURRENT_PACKAGES_DIR}/debug/share
-    ${CURRENT_PACKAGES_DIR}/bin/sdl2-config
-    ${CURRENT_PACKAGES_DIR}/debug/bin/sdl2-config
-    ${CURRENT_PACKAGES_DIR}/SDL2.framework
-    ${CURRENT_PACKAGES_DIR}/debug/SDL2.framework
+    "${CURRENT_PACKAGES_DIR}/debug/include"
+    "${CURRENT_PACKAGES_DIR}/debug/share"
+    "${CURRENT_PACKAGES_DIR}/bin/sdl2-config"
+    "${CURRENT_PACKAGES_DIR}/debug/bin/sdl2-config"
+    "${CURRENT_PACKAGES_DIR}/SDL2.framework"
+    "${CURRENT_PACKAGES_DIR}/debug/SDL2.framework"
 )
 
-file(GLOB BINS ${CURRENT_PACKAGES_DIR}/debug/bin/* ${CURRENT_PACKAGES_DIR}/bin/*)
+file(GLOB BINS "${CURRENT_PACKAGES_DIR}/debug/bin/*" "${CURRENT_PACKAGES_DIR}/bin/*")
 if(NOT BINS)
     file(REMOVE_RECURSE
-        ${CURRENT_PACKAGES_DIR}/bin
-        ${CURRENT_PACKAGES_DIR}/debug/bin
+        "${CURRENT_PACKAGES_DIR}/bin"
+        "${CURRENT_PACKAGES_DIR}/debug/bin"
     )
 endif()
 
-if(NOT VCPKG_CMAKE_SYSTEM_NAME)
+if(VCPKG_TARGET_IS_WINDOWS)
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-        file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/lib/manual-link)
-        file(RENAME ${CURRENT_PACKAGES_DIR}/lib/SDL2main.lib ${CURRENT_PACKAGES_DIR}/lib/manual-link/SDL2main.lib)
+        file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/lib/manual-link")
+        file(RENAME "${CURRENT_PACKAGES_DIR}/lib/SDL2main.lib" "${CURRENT_PACKAGES_DIR}/lib/manual-link/SDL2main.lib")
     endif()
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
-        file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link)
-        file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/SDL2maind.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/SDL2maind.lib)
+        file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link")
+        file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/SDL2maind.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/SDL2maind.lib")
     endif()
 
-    file(GLOB SHARE_FILES ${CURRENT_PACKAGES_DIR}/share/sdl2/*.cmake)
+    file(GLOB SHARE_FILES "${CURRENT_PACKAGES_DIR}/share/sdl2/*.cmake")
     foreach(SHARE_FILE ${SHARE_FILES})
         vcpkg_replace_string("${SHARE_FILE}" "lib/SDL2main" "lib/manual-link/SDL2main")
     endforeach()
 endif()
 
-configure_file(${SOURCE_PATH}/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
+configure_file("${SOURCE_PATH}/LICENSE.txt" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
 vcpkg_copy_pdbs()
 
 set(DYLIB_COMPATIBILITY_VERSION_REGEX "set\\(DYLIB_COMPATIBILITY_VERSION (.+)\\)")

--- a/ports/sdl2/vcpkg.json
+++ b/ports/sdl2/vcpkg.json
@@ -1,12 +1,25 @@
 {
   "name": "sdl2",
   "version": "2.0.16",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.",
   "homepage": "https://www.libsdl.org/download-2.0.php",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
   "features": {
     "vulkan": {
       "description": "Vulkan functionality for SDL"
+    },
+    "x11": {
+      "description": "Dynamically load X11 support"
     }
   }
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6098,7 +6098,7 @@
     },
     "sdl2": {
       "baseline": "2.0.16",
-      "port-version": 1
+      "port-version": 2
     },
     "sdl2-gfx": {
       "baseline": "1.0.4",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2130,7 +2130,7 @@
     },
     "ffmpeg": {
       "baseline": "4.4",
-      "port-version": 17
+      "port-version": 18
     },
     "ffnvcodec": {
       "baseline": "11.1.5.0",
@@ -2326,7 +2326,7 @@
     },
     "gamedev-framework": {
       "baseline": "0.20.0",
-      "port-version": 0
+      "port-version": 1
     },
     "gamenetworkingsockets": {
       "baseline": "1.3.0",
@@ -5190,7 +5190,7 @@
     },
     "pixel": {
       "baseline": "0.3",
-      "port-version": 2
+      "port-version": 3
     },
     "pixman": {
       "baseline": "0.40.0",

--- a/versions/f-/ffmpeg.json
+++ b/versions/f-/ffmpeg.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "fa1d927c640e5fc9abb088c7409e90057cd0b371",
+      "version": "4.4",
+      "port-version": 18
+    },
+    {
       "git-tree": "795e31317ef58fbe13247af9e8bade71f3edde5e",
       "version": "4.4",
       "port-version": 17

--- a/versions/g-/gamedev-framework.json
+++ b/versions/g-/gamedev-framework.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a9771cd4f7093f5d34a78d3e2417707185444910",
+      "version-semver": "0.20.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "195d4f41db9d03aa44fa6cc0d984668115a17718",
       "version-semver": "0.20.0",
       "port-version": 0

--- a/versions/p-/pixel.json
+++ b/versions/p-/pixel.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "57198090e451e125ad231a0a1068643cbd06cb61",
+      "version-string": "0.3",
+      "port-version": 3
+    },
+    {
       "git-tree": "bd45dec4585e8be40088a7017dcce546a009748f",
       "version-string": "0.3",
       "port-version": 2

--- a/versions/s-/sdl2.json
+++ b/versions/s-/sdl2.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "946628e2d440c855802b11384b6875633b7d642d",
+      "git-tree": "120b4a3f60f1b2bfc0f20a7c20b5494fff8ea519",
       "version": "2.0.16",
       "port-version": 2
     },

--- a/versions/s-/sdl2.json
+++ b/versions/s-/sdl2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "946628e2d440c855802b11384b6875633b7d642d",
+      "version": "2.0.16",
+      "port-version": 2
+    },
+    {
       "git-tree": "f39203f93b1c068fca2dd7b7501891d8aca3b65d",
       "version": "2.0.16",
       "port-version": 1

